static-analysis/teammates-macker.xml

```diff
@@ -1,45 +1,399 @@
+                        <to class="${storage}.entity.**" />
+                    </allow>
+                    <allow>
+                        <from>
+                            <!-- Violations that need fixing -->
+                            <include class="${common}.datatransfer.questions.FeedbackConstantSumQuestionDetails" />
+                            <include class="${common}.datatransfer.questions.FeedbackMcqQuestionDetails" />
+                            <include class="${common}.datatransfer.questions.FeedbackMsqQuestionDetails" />
+                            <include class="${common}.datatransfer.questions.FeedbackRankOptionsQuestionDetails" />
+                            <include class="${common}.datatransfer.questions.FeedbackRankRecipientsQuestionDetails" />
+                            <include class="${common}.datatransfer.questions.FeedbackRankRecipientsQuestionDetails**$**" />                        
+                        </from>
+                    </allow>
+                </deny>
+            </access-rule>
+        </ruleset>
+        
+        <ruleset name="UiComponent: Only *Action can touch Logic API">
+            <access-rule>
+                <deny>
+                    <from class="${ui}.**" />
+                    <to class="${logic}.**" />   
+                    <allow>
+                        <from>
+                            <include class="${ui}.controller.**" />
+                            <include class="${ui}.automated.**" />
+                        </from>
+                        <to class="${logic}.api.**" />
+                    </allow>
+                    <allow>
+                        <!-- to import sample data, need to be changed -->
+                        <from class="${ui}.controller.AdminInstructorAccountAddAction" />
+                        <to class="${logic}.backdoor.BackDoorLogic" />
+                    </allow>    
+                </deny>
+            </access-rule>
+        </ruleset>
+        <ruleset name="UiComponent: Templates are only used in Pagedata">
+            <access-rule>
+                <deny>
+                    <from class="${ui}.**" >
+                        <exclude class="${ui}.template.**" /> <!-- allow dependency within the package -->
+                    </from>
+                    <to class="${ui}.template.**" />
+                    <allow>
+                        <from>
+                            <include class="${ui}.pagedata.**" />
+                            
+                            <!-- Violations that need fixing -->
+                            <include class="${common}.datatransfer.questions.FeedbackRankOptionsQuestionDetails" />
+                            <include class="${common}.datatransfer.questions.FeedbackRankRecipientsQuestionDetails**$**" />
+                            <include class="${common}.datatransfer.questions.FeedbackRankRecipientsQuestionDetails" />
```

Comment:
I don't think the above three are needed. Packages outside `${ui}` should not be relevant in this ruleset.